src/placeos-models/user.cr

```diff
@@ -167,14 +167,15 @@ module PlaceOS::Model
 
     secondary_index :email_digest
 
-    def self.find_by_email(authority_id : String, email : String)
+    def self.find_by_email(authority_id : String, email : PlaceOS::Model::Email)
       find_by_emails(authority_id, [email]).first?
     end
 
-    def self.find_by_emails(authority_id : String, emails : Enumerable(String))
+    def self.find_by_emails(authority_id : String, emails : Array(PlaceOS::Model::Email))
       return [] of self if emails.empty?
 
-      digests = emails.map &->weak_digest(String)
+      digests = emails.map { |email| weak_digest(email.to_s) }
```

Comment:
This digest should be in the email struct

Reply:
It's already in the struct, so code related to it in the User model can be removed, and this line of code simplified to
```suggestion
    digests = emails.map &.digest
```